File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/extending-word-documents-and-excel-workbooks-in-vsto-add-ins-at-run-time_6.vb
    Private document2 As Word.Document = Nothing
    Private WithEvents vstoDocument As Microsoft.Office.Tools.Word.Document = Nothing

    Private Sub CreateDocument2()
        document2 = Me.Application.Documents.Add()
        vstoDocument = Globals.Factory.GetVstoObject(document2)
    End Sub

    Private Sub vstoDocument_BeforeSave(ByVal sender As Object, _
        ByVal e As SaveEventArgs) Handles vstoDocument.BeforeSave
        e.Cancel = True
    End Sub

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/extending-word-documents-and-excel-workbooks-in-vsto-add-ins-at-run-time_6.cs
        private Word.Document document2 = null;
        private Microsoft.Office.Tools.Word.Document vstoDocument = null;

        private void CreateDocument2()
        {
            document2 = this.Application.Documents.Add(ref missing,
                ref missing, ref missing, ref missing);
            vstoDocument = Globals.Factory.GetVstoObject(document2);
            vstoDocument.BeforeSave += new SaveEventHandler(vstoDocument_BeforeSave);
        }

        private void vstoDocument_BeforeSave(object sender, SaveEventArgs e)
        {
            e.Cancel = true;
        }